src/AutorestSwift/View Models/RequestViewModel.swift

```diff
@@ -58,8 +58,15 @@ struct RequestViewModel {
         }
         self.params = params
 
-        self.objectType = request.signatureParameters?.first?.schema.name
-        self.objectName = request.signatureParameters?.first?.name
+        // TODO: only support the first signature parameter in Reqest now
+        let firstSignatreParam = request.signatureParameters?.first
+        self.objectType = firstSignatreParam?.schema.name
+        if firstSignatreParam?.required ?? false {
+            self.objectName = firstSignatreParam?.name
+        } else {
+            let optionName = "\(operation.name)Options?".toCamelCase
+            self.objectName = "\(optionName).\(firstSignatreParam?.name ?? "")"
+        }
```

Comment:
There's a typo here (signatre => signature) but I'm not understanding the comment. What is special about the first signature parameter? Why would the options object ever not have the same name as the method?

Reply:
We only support 1  request object (the class which we json encode to build the http payload) now  and the signure parameter is used to built the objectName of the request object ` self.objectName = firstSignatreParam?.name` . Currently 'self.objectName' is a String (not an array), that is why we only take the first signature parameters for the objectName.

Reply:
So we are looking for the body parameter? I think it would make sense to iterate through the signatureParameters and filter by those that have `http.in = .body`. We should then assert there is only 1 and continue. Instead of calling it `firstSignatureParam` I would call it "bodyParam" and put a comment that we are looking for a body-encoded parameter, if any.